### DISCUSSION TITLE
[firebase_messaging] why data message on android is not supported by plugin, message is lost when app terminated

### DISCUSSION THREAD

**Discussion by tbm98** [firebase/flutterfire#5595] [Feature request] [answered]
Why does it work on iOS but not Android? whether it works in the future or not?

**Comment by iapicca** [comment]
Hi @tbm98 
can you please provide your `flutter doctor -v` , 
your `flutter run --verbose` 
and your `pubspec.yaml` 
or if possible a reproducible minimal code sample.
Thank you

**Comment by tbm98** [comment]
<img width="664" alt="Screen Shot 2020-03-09 at 22 49 52" src="https://user-images.githubusercontent.com/52562340/76231813-5b804b00-6258-11ea-841c-01014653ae78.png">
this picture is showing data message on android when app terminated is not support by plugin, message is lost

**Comment by xyeeeChen** [comment]
In my test, the data message on Android is:

App in Background: `onBackgroundMessage` is called, not `onMessage`
App Terminated: `onBackgroundMessage` is called, not `without support`

**Comment by tbm98** [comment]
> In my test, the data message on Android is:
> 
> App in Background: `onBackgroundMessage` is called, not `onMessage`
> App Terminated: `onBackgroundMessage` is called, not `without support`

But why  in official docs still showing `not supported by plugin, message is lost`

**Comment by tbm98** [comment]
[here](https://pub.dev/packages/firebase_messaging#receiving-messages)

**Comment by nicksav** [comment]
Ok any luck guys? Is it working or not?

**Comment by Yirmeyah-d** [comment]
Any luck Guys ? did someone try to use onBackgroundmessage ?

**Comment by nicksav** [comment]
Hey yes.
Its working now!!!

**Comment by Yirmeyah-d** [comment, marked as answer]
Oh really ? 
Could you show me how you succeed ?
I'm actually struggling with some issue ?  https://github.com/FirebaseExtended/flutterfire/issues/2565
Could you help me ?
